### PR TITLE
MB-15167 - Add weight displays to review shipment weights

### PR DESCRIPTION
**Pull request by michaelvaldes-truss**
## Summary

This PR should add weight displays to the review shipment weights page.

Things to note: 
1. I reused the useCalculatedEstimatedWeight hook in a few components, including MoveDetails.jsx where I slightly changed how the calculation works. Previously, it only considered shipments that had the APPROVED status, and the hook it's now using also accounts for shipments with a few other statuses. This was done to ensure that the calculation works the same on both the MoveDetails and MoveTaskOrder pages.
2. I couldn't determine whether I need to add any alerts at the top or include any links in the weight displays.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the office app as a services counselor
3. Navigate to a move
4. Navigate to the weights page by appending 'review-shipment-weights' to the url after the move code
5. Ensure that for moves with a risk of excess weight (e.g. move code RISKEX), there is a corresponding tag in the total estimated weight display.
6. Ensure that for moves with external vendor shipments (e.g. move code PRXNTS), there is a corresponding tag in the estimated weight display with a link to the move details page.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

## Screenshots
image

### CODE REVIEW

File: src/pages/Office/ServicesCounselingMoveInfo/ServicesCounselingMoveInfo.test.jsx [hunk 1]
```diff
@@ -146,6 +84,76 @@ const newMoveDetailsQuery = {
...
+    1: {
+      id: '1',
+      originDutyLocation: {
+        address: {
+          streetAddress1: '',
+          city: 'Fort Knox',
+          state: 'KY',
+          postalCode: '40121',
+        },
+      },
```
Comment (ruizajtruss) [inline src/pages/Office/ServicesCounselingMoveInfo/ServicesCounselingMoveInfo.test.jsx:97]: **Praise:** Appreciate the re-ordering here!

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx [hunk 2]
```diff
@@ -117,8 +116,8 @@ export const MoveTaskOrder = ({ match, ...props }) => {
   } = props;
 
   const { orders = {}, move, mtoShipments, mtoServiceItems, isLoading, isError } = useMoveTaskOrderQueries(moveCode);
-
   const order = Object.values(orders)?.[0];
+  const estimatedWeightTotal = useCalculatedEstimatedWeight(mtoShipments);
```
Comment (ruizajtruss) [inline src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx:120]: **Question:** I wonder the same for the calculation here

File: src/pages/Office/MoveDetails/MoveDetails.jsx [hunk 3]
```diff
@@ -61,6 +62,8 @@ const MoveDetails = ({
 
   const { move, order, mtoShipments, mtoServiceItems, isLoading, isError } = useMoveDetailsQueries(moveCode);
 
+  const estimatedWeight = useCalculatedEstimatedWeight(mtoShipments);
```
Comment (ruizajtruss) [inline src/pages/Office/MoveDetails/MoveDetails.jsx:65]: **Question:** I'm curious if we should keep this calculation function within the useEffect on line 174 to ensure it runs on first render and every following update to the values in the dependency array

File: src/hooks/custom.js [hunk 4]
```diff
@@ -38,10 +38,13 @@ export const useCalculatedWeightRequested = (mtoShipments) => {
...
-        return prev + current.primeEstimatedWeight;
-      }, 0);
+    if (mtoShipments?.some((s) => includedStatusesForCalculatingWeights(s.status) && s.primeEstimatedWeight)) {
+      return mtoShipments
+        ?.filter((s) => includedStatusesForCalculatingWeights(s.status) && s.primeEstimatedWeight)
+        .reduce((prev, current) => {
+          return prev + current.primeEstimatedWeight;
+        }, 0);
+    }
+    return null;
```
Comment (ruizajtruss) [inline src/hooks/custom.js:48]: **Praise:** Love the new approach here, saves us time incase none of the weights have the status we're looking for

**Review by ruizajtruss — CHANGES_REQUESTED** [review]
LGTM! 🚀  Just one question surrounding useCalculatedEstimatedWeight

File: src/pages/Office/MoveDetails/MoveDetails.jsx (hunk 3 above)
Comment (michaelvaldes-truss) [reply src/pages/Office/MoveDetails/MoveDetails.jsx]: Good point -- I changed it back to useEffect in all the places where we were doing that previously.

**Review by ruizajtruss — APPROVED** [review]
Looks great! 🥳

**Comment by AnnaGingle** [comment]
[_*I am unable to review this PR because of my OCONUS OOO but the platform team is working on that._ 

**The page is off to a great start but I see a couple of issues:**

- weightDisplay grid should fill the 960px container
- the 4 weightDisplay components should have a width of 222px
- Estimated weight (total) weightDisplay is larger than the others. The content should have a carriage return see [Figma](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=8643%3A100689&t=4DjUrz8HtrGNEN71-4)](https://github.com/transcom/mymove/pull/10149)

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx [hunk 5]
```diff
@@ -725,8 +726,8 @@ export const MoveTaskOrder = ({ match, ...props }) => {
                 externalVendorShipmentCount > 0 && <br />}
               {externalVendorShipmentCount > 0 && (
                 <small>
-                  {externalVendorShipmentCount} shipment not moved by GHC prime.{' '}
-                  <Link className="usa-link" to={`/moves/${moveCode}`}>
+                  {externalVendorShipmentCount} shipment{externalVendorShipmentCount > 1 && 's'} not moved by GHC prime.{' '}
```
Comment (ruizajtruss) [inline src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx:729]: **Praise:** Clever touch!

**Review by ruizajtruss — APPROVED** [review]
LGTM 🚀

**Comment by AnnaGingle** [comment]
The external vendor text should show in the estimated weight (total) weightDisplay if a move has a PPM because it is not moved by the GHC prime

**Review by AnnaGingle — CHANGES_REQUESTED** [review]
The external vendor text should show in the estimated weight (total) weightDisplay if a move has a PPM because it is not moved by the GHC prime
Figma link https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=5689%3A80776&t=59vrz5pcygurBmyD-4

**Review by AnnaGingle — APPROVED** [review]
Great job!!!

**Review by dinabeigelman — APPROVED** [review]
pushing the approve button for @AnnaGingle

**Pull request merged**